api/v1alpha1/underlay_types.go

```diff
@@ -22,6 +22,12 @@ import (
 
 // UnderlaySpec defines the desired state of Underlay.
 type UnderlaySpec struct {
+	// NodeSelector specifies which nodes this Underlay applies to.
+	// If empty or not specified, applies to all nodes (backward compatible).
+	// Multiple Underlays with overlapping node selectors will be rejected.
```

Comment:
Question: so if there's an overlap in *a single* node we reject the entire underlay ? or it is usable - but not on the node w/ the overlap ? 

I do not remember what we've defined in the enhancement. It would be nice to link it here.